### PR TITLE
Add more test for Pub Sub cloud explorer resources.

### PR DESCRIPTION
**Pull request by ILMTitan**

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ISourceRootViewModelBase.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        /// Returns the tree node to use while loading data.
+        /// </summary>
+        TreeLeaf LoadingPlaceholder { get; }
+
+        /// <summary>
+        /// Returns the context in which this source root view model is working.
+        /// </summary>
+        ICloudSourceContext Context { get; }
+
+        void Initialize(ICloudSourceContext context);
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ISourceRootViewModelBase.cs]: Comments on the methods as well.

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeHierarchy.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using System.Collections.ObjectModel;
+
+namespace GoogleCloudExtension.CloudExplorer
+{
+    public interface ITreeHierarchy : ITreeNode
+    {
+        /// <summary>
+        /// The children for this item.
+        /// </summary>
+        ObservableCollection<TreeNode> Children { get; }
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeHierarchy.cs]: For the interface you probably want to restrict yourself to the `IList<T>` interface. The implementations should use the `ObservableCollection<T>`.

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeNode.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.ComponentModel;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+namespace GoogleCloudExtension.CloudExplorer
+{
+    public interface ITreeNode : INotifyPropertyChanged
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeNode.cs]: The interfaces `ITreeNode` and `INotifyPropertyChanged` are really orthogonal. The implementations of `ITreeNode` most probably do also implement `INotifyPropertyChanged` but these interfaces are not related.

Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeNode.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ISourceRootViewModelBase.cs (hunk 1 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ISourceRootViewModelBase.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeHierarchy.cs (hunk 2 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/ITreeHierarchy.cs]: Because C# properties are not covariant, this means the `TreeHierarchy` class will also need to change. This doe not seem to cause any problems.

**Comment by ILMTitan** [comment]
PTAL.
